=== modified file 'analytic/analytic.py'
--- analytic/analytic.py        2011-01-18 17:31:53 +0000
+++ analytic/analytic.py        2013-11-04 11:03:50 +0000
@@ -250,7 +250,7 @@
         'unit_amount': fields.float('Quantity', help='Specifies the amount of quantity to count.'),
         'account_id': fields.many2one('account.analytic.account', 'Analytic Account', required=True, ondelete='cascade', select=True, domain=[('type','<>','view')]),
         'user_id': fields.many2one('res.users', 'User'),
-        'company_id': fields.related('account_id', 'company_id', type='many2one', relation='res.company', string='Company', store=True, readonly=True),
+        'company_id': fields.many2one('res.company', 'Company', readonly=True),
 
     }
     _defaults = {
