=== modified file 'analytic/analytic.py'
--- analytic/analytic.py    2011-01-18 17:31:53 +0000
+++ analytic/analytic.py    2014-04-25 14:37:08 +0000
@@ -51,6 +51,8 @@
         if context is None:
             context = {}
         child_ids = tuple(self.search(cr, uid, [('parent_id', 'child_of', ids)]))
+        user = self.pool.get('res.users').browse(cr, uid, uid)
+        company_ids = tuple(self.pool.get('res.company').search(cr, uid, [('id', 'child_of', [user.company_id.id])]))
         for i in child_ids:
             res[i] =  {}
             for n in name:
@@ -60,7 +62,7 @@
             return res

         where_date = ''
-        where_clause_args = [tuple(child_ids)]
+        where_clause_args = [company_ids,tuple(child_ids)]
         if context.get('from_date', False):
             where_date += " AND l.date >= %s"
             where_clause_args  += [context['from_date']]
@@ -85,7 +87,7 @@
                      COALESCE(SUM(l.unit_amount),0) AS quantity
               FROM account_analytic_account a
                   LEFT JOIN account_analytic_line l ON (a.id = l.account_id)
-              WHERE a.id IN %s
+              WHERE l.company_id in %s and a.id IN %s
               """ + where_date + """
               GROUP BY a.id""", where_clause_args)
         for ac_id, debit, credit, balance, quantity in cr.fetchall():

