=== modified file 'base_vat_unique/base_vat_unique.py'
--- base_vat_unique/base_vat_unique.py  2011-04-27 18:03:39 +0000
+++ base_vat_unique/base_vat_unique.py  2014-07-21 16:27:33 +0000
@@ -27,9 +27,15 @@
     _inherit = "res.partner"

     _sql_constraints = [
-        ('vat_uniq', 'unique (vat)', _('Error! Specified VAT Number already exists for any other registered partner.'))
+        ('vat_uniq', 'unique (vat,active)', _('Error! Specified VAT Number already exists for any other registered partner.'))
     ]

+    def copy(self, cr, uid, ids, default=None, context=None):
+        if default is None:
+            default = {}
+        default['active'] = True
+        return super(res_partner, self).copy(cr, uid, ids, default=default, context=context)
+
 res_partner()

 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:

