=== modified file 'account/invoice.py'
--- account/invoice.py  2011-02-28 06:36:59 +0000
+++ account/invoice.py  2014-07-21 16:18:19 +0000
@@ -1653,6 +1653,16 @@
         'invoice_ids': fields.one2many('account.invoice.line', 'partner_id', 'Invoices', readonly=True),
     }

+    def copy(self, cr, uid, id, default=None, context=None):
+        if default is None:
+            default = {}
+
+        if context is None:
+            context = {}
+
+        default.update({'invoice_ids' : []})
+        return super(res_partner, self).copy(cr, uid, id, default, context)
+
 res_partner()

 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
