=== modified file 'l10n_es_account_asset/account_asset.py'
--- l10n_es_account_asset/account_asset.py      2013-01-03 19:14:54 +0000
+++ l10n_es_account_asset/account_asset.py      2015-07-27 11:27:13 +0000
@@ -134,9 +134,9 @@
                 end_date = end_date + relativedelta(months=+asset.method_number)
         last_depreciation_date = asset._get_last_depreciation_date()
         if last_depreciation_date:
-            last_depreciation_date = datetime.strptime(last_depreciation_date[asset.id], '%Y-%m-%d') 
+            last_depreciation_date = datetime.strptime(last_depreciation_date[asset.id], '%Y-%m-%d')
         undone_dotation_number = 0
-        while depreciation_date <= end_date or (depreciation_date.year == end_date.year and asset.prorata and asset.method_period == 12):
+        while depreciation_date < end_date or (depreciation_date.year == end_date.year and asset.prorata and asset.method_period == 12):
             depreciation_date = (datetime(depreciation_date.year, depreciation_date.month, depreciation_date.day) + relativedelta(months=+asset.method_period))
             if last_depreciation_date:
                 if depreciation_date > last_depreciation_date:
