=== modified file 'stock/product.py'
--- stock/product.py	2011-04-08 13:32:12 +0000
+++ stock/product.py	2012-03-26 18:48:00 +0000
@@ -190,6 +190,13 @@
         if not ids:
             return res
 
+        if context.get('company_id', False):
+            company_id = context['company_id']
+        else:
+            company_id = self.pool.get('res.users').browse(cr, uid, uid).company_id.id
+
+        company_id = self.pool.get('res.company').search(cr, uid, [('id', 'child_of', [company_id])])
+
     # TODO: write in more ORM way, less queries, more pg84 magic
         if context.get('shop', False):
             cr.execute('select warehouse_id from sale_shop where id=%s', (int(context['shop']),))
@@ -262,7 +269,7 @@
                 'and location_dest_id IN %s '\
                 'and product_id IN %s '\
                 '' + (prodlot_id and ('and prodlot_id = ' + str(prodlot_id)) or '') + ' '\
-                'and state IN %s ' + (date_str and 'and '+date_str+' ' or '') +' '\
+                'and state IN %s ' + (date_str and 'and '+date_str+' ' or '') +' and company_id IN (' + ",".join([str(x) for x in company_id]) + ') '\
                 'group by product_id,product_uom',tuple(where))
             results = cr.fetchall()
         if 'out' in what:
@@ -274,7 +281,7 @@
                 'and location_dest_id NOT IN %s '\
                 'and product_id  IN %s '\
                 '' + (prodlot_id and ('and prodlot_id = ' + str(prodlot_id)) or '') + ' '\
-                'and state in %s ' + (date_str and 'and '+date_str+' ' or '') + ' '\
+                'and state in %s ' + (date_str and 'and '+date_str+' ' or '') + ' and company_id IN (' + ",".join([str(x) for x in company_id]) + ') '\
                 'group by product_id,product_uom',tuple(where))
             results2 = cr.fetchall()
         uom_obj = self.pool.get('product.uom')
