=== modified file 'account/invoice.py'
--- account/invoice.py	2012-08-20 15:53:34 +0000
+++ account/invoice.py	2012-10-25 09:18:49 +0000
@@ -1286,9 +1286,9 @@
         'uos_id': fields.many2one('product.uom', 'Unit of Measure', ondelete='set null'),
         'product_id': fields.many2one('product.product', 'Product', ondelete='set null'),
         'account_id': fields.many2one('account.account', 'Account', required=True, domain=[('type','<>','view'), ('type', '<>', 'closed')], help="The income or expense account related to the selected product."),
-        'price_unit': fields.float('Unit Price', required=True, digits_compute= dp.get_precision('Account')),
+        'price_unit': fields.float('Unit Price', required=True, digits = (12,3)),
         'price_subtotal': fields.function(_amount_line, method=True, string='Subtotal', type="float",
-            digits_compute= dp.get_precision('Account'), store=True),
+            digits = (12,3), store=True),
         'quantity': fields.float('Quantity', required=True),
         'discount': fields.float('Discount (%)', digits_compute= dp.get_precision('Account')),
         'invoice_line_tax_id': fields.many2many('account.tax', 'account_invoice_line_tax', 'invoice_line_id', 'tax_id', 'Taxes', domain=[('parent_id','=',False)]),

=== modified file 'purchase/purchase.py'
--- purchase/purchase.py	2012-06-29 14:14:29 +0000
+++ purchase/purchase.py	2012-10-25 09:19:52 +0000
@@ -632,8 +632,8 @@
         'product_id': fields.many2one('product.product', 'Product', domain=[('purchase_ok','=',True)], change_default=True),
         'move_ids': fields.one2many('stock.move', 'purchase_line_id', 'Reservation', readonly=True, ondelete='set null'),
         'move_dest_id': fields.many2one('stock.move', 'Reservation Destination', ondelete='set null'),
-        'price_unit': fields.float('Unit Price', required=True, digits_compute= dp.get_precision('Purchase Price')),
-        'price_subtotal': fields.function(_amount_line, method=True, string='Subtotal', digits_compute= dp.get_precision('Purchase Price')),
+        'price_unit': fields.float('Unit Price', required=True, digits = (12,3)),
+        'price_subtotal': fields.function(_amount_line, method=True, string='Subtotal', digits = (12,3)),
         'notes': fields.text('Notes'),
         'order_id': fields.many2one('purchase.order', 'Order Reference', select=True, required=True, ondelete='cascade'),
         'account_analytic_id':fields.many2one('account.analytic.account', 'Analytic Account',),

