=== modified file 'addons/openerp/controllers/impex.py'
--- addons/openerp/controllers/impex.py 2012-07-30 11:08:14 +0000
+++ addons/openerp/controllers/impex.py 2013-03-12 11:40:10 +0000
@@ -120,10 +119,15 @@
             for i, view in enumerate(params.view_mode):
                 views[view] = params.view_ids[i]
 
-
+        
         exports = rpc.RPCProxy('ir.exports')
 
         headers = [{'string' : 'Name', 'name' : 'name', 'type' : 'char'}]
+        if isinstance(import_compat, (str, unicode)):
+            import_compat = int(import_compat)
+        else:
+            import_compat = int(import_compat[0])
+        
         tree = treegrid.TreeGrid('export_fields',
                                  model=params.model,
                                  headers=headers,
@@ -131,7 +135,7 @@
                                  field_parent='relation',
                                  context=ctx,
                                  views=views,
-                                 import_compat=int(import_compat))
+                                 import_compat=import_compat)
 
         tree.show_headers = False
 
=== modified file 'addons/openerp/controllers/templates/exp.mako'
--- addons/openerp/controllers/templates/exp.mako       2011-01-17 13:17:07 +0000
+++ addons/openerp/controllers/templates/exp.mako       2013-03-12 11:11:47 +0000
@@ -167,9 +167,7 @@
                         <td>
                             <select id="import_compat" name="import_compat" onchange="do_import_cmp();">
                                 <option value="1">${_("Import Compatible Export")}</option>
-                                <option value="0"
-                                    ${'selected=selected' if import_compat == "0" else ''}
-                                    >${_("Export all Data")}</option>
+                                <option value="0">${_("Export all Data")}</option>
                             </select>
                         </td>
                     </tr>

